### DISCUSSION TITLE
config keymap for treesitter

### DISCUSSION THREAD

**Discussion by shelper** [LazyVim/LazyVim#2444] [Q&A]
i am trying to set the treesitter keymap to replace '<C-space>' with '<C-x>', and '<BS>' with '<C-->', i did below but seems it does not work

1. add treesitter.lua to my plugins folder. and lines below:

```
return {
    "nvim-treesitter/nvim-treesitter",
    keys = {
      { "<C-x>", desc = "Increment selection" },
      { "<C-->", desc = "Decrement selection", mode = "x" },
    },
  }
}
```

after restarting vim, none of this keymaps is in the map list ..

**Comment by kobzar** [comment]
If you mean that Decrement dont work try to ecran the dash symbol like "C-\-" or "C-\\-"
But I am not shure.

**Reply by shelper** [reply]
Ctrl + '-', is what i mean, but C-x also is not working

**Comment by dpetka2001** [comment]
Try this
```lua
return {
  {
    "nvim-treesitter/nvim-treesitter",
    opts = {
      incremental_selection = {
        keymaps = {
          init_selection = "<C-x>",
          node_incremental = "<C-x>",
          node_decremental = "<C-->",
        },
      },
    },
  },
}
```
Make sure the keys you want to use are available.

**Reply by cohml** [reply]
But how can one set a `desc` for these mappings? e.g., for nice visuals via `which-key`. These parameters seem to expect a string, so passing tables with `desc` raises an error.

**Reply by dpetka2001** [reply]
They get automatically descriptions from `nvim-treesitter`, since they're internal mappings. Just press `<leader>` and then `backspace` and you will see them. They're just not available in the `<leader>` top-level mappings.

**Reply by cohml** [reply]
Hm. I wanted to override the default descriptions. So rather than set the mappings via `setup`, I just call the following after `setup` returns, which seems to work as intended (NB: I don't use scope):

```lua
local incremental_selection = require('nvim-treesitter.incremental_selection')
vim.keymap.set("n", "<leader>ss", incremental_selection.init_selection, { desc = "[S]tart Selection" })
vim.keymap.set("v", "<leader>si", incremental_selection.node_incremental, { desc = "[I]ncrement" })
vim.keymap.set("v", "<leader>sd", incremental_selection.node_decremental, { desc = "[D]ecrement" })
```

Just in case anyone else has a similar desire in the future.
